### PR TITLE
Unify (app) banners

### PR DESCRIPTION
**Pull request by nemanjaglumac**
> [!IMPORTANT]
> Prerequisite for #49344

### What does this PR accomplish?
- It introduces some order in the components that rely on the `Banner` component.
- It uses the icons defined in our design system, and not emojis like before
- It aligns the icon and the beginning of text with the hamburger menu and the rest of the icons in sidebar
- Slightly increases the contrast of the Payment banner by using a bit darker shade of the background color

## Before
Screenshot 2024-11-21 at 19 39 09
Screenshot 2024-11-21 at 19 38 50
Screenshot 2024-11-21 at 19 38 38
Screenshot 2024-11-21 at 19 38 06

## Now
Screenshot 2024-11-21 at 19 36 07
Screenshot 2024-11-21 at 19 36 36
Screenshot 2024-11-21 at 19 36 58
Screenshot 2024-11-21 at 19 37 44

### CODE REVIEW

File: frontend/src/metabase/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -1,19 +1,48 @@
-import Markdown from "metabase/core/components/Markdown";
-import { Flex, type FlexProps } from "metabase/ui";
+import type { ReactNode } from "react";
 
-export const Banner = ({ children, ...props }: FlexProps) => {
-  const content =
-    typeof children === "string" ? <Markdown>{children}</Markdown> : children;
```
Comment (nemanjaglumac) [inline frontend/src/metabase/components/Banner/Banner.tsx:6]: This dance between "regular" children or the string rendered in `Markdown` is not used anywhere else. It has been used only in one random place (up until https://github.com/metabase/metabase/pull/49273, I believe).

No need to continue supporting that pattern, imo.

File: frontend/src/metabase/components/Banner/Banner.tsx [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
-      bg="bg-light"
-      c="text-medium"
       data-testid="app-banner"
-      p="0.75rem"
-      {...props}
+      align="center"
+      bg={bg || "bg-medium"}
+      py="sm"
+      justify="space-between"
+      pl="1.325rem"
```
Comment (heypoom) [inline frontend/src/metabase/components/Banner/Banner.tsx:31]: `1.325rem` seems quite specific here

**Review by heypoom — APPROVED** [review]
The new design looks great!

File: frontend/src/metabase/components/Banner/Banner.tsx (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/components/Banner/Banner.tsx:31]: Yes, unfortunately we're doing silly things in the main nav sidebar.
it has its own padding, and then each menu item has its own padding, and then each icon in the menu item has... you guessed it  - its own padding.

So... 🤷‍♂️

**Pull request merged**